src/components/TestRail/TestRail.js

```diff
@@ -196,10 +241,18 @@ class TestRail {
             if (result.hasElapsedTime()) {
                 resultEntry.elapsed = result.getElapsed();
             }
-            
-            postData.results.push(resultEntry);
+    
+            if (includeCasesOnlyIfExistInRun) { // Check if the case_id is valid
+                if (validCaseIds.includes(parseInt(resultEntry.case_id))) { // Check if the case_id is valid
+                    postData.results.push(resultEntry);
+                } else {
+                    ColorConsole.error(`Test case C${resultEntry.case_id} is not valid for run R${runID}. Skipping.`);
+                }
+            } else {
+                postData.results.push(resultEntry); // Add all results, regardless of validity
```

Comment:
You can avoid having `postData.results.push(resultEntry);` twice:
```suggestion
            if (includeCasesOnlyIfExistInRun && !validCaseIds.includes(parseInt(resultEntry.case_id))) { // Check if the case_id is valid
              ColorConsole.error(`Test case C${resultEntry.case_id} is not valid for run R${runID}. Skipping.`);
            } else {
                postData.results.push(resultEntry); // Add the result
```

Reply:
nice